### ISSUE TITLE
[FEATURE] Remove mandatory dependency of Hadoop environment

### ISSUE THREAD

**Issue by wolfboys** [streamxhub/streamx#233]
At present, streamx only support "yarn application" and must be on Hadoop environment. In order to be compatible with k8s and local deployment, we need to remove the mandatory dependency of Hadoop environment and make it optional

There are about three steps to be improved:

1. Remove all operations of uploading files to HDFS during service startup initialization (Flink, streamx supports related jars)

2. If the new task is in "yarn application" mode, need to upload the relevant jar files supported by Flink and streamx to HDFS (only for the first yarn application job)

3. For the Flink SQL task, the jar file and the jar file downloaded by Maven POM select whether to upload HDFS according to the deployment mode

**Comment by Al-assad** [comment]
@wolfboys What's more, I think we should further abstract the file system used by StreamX workspace which storing temporary files, compiled flink job jar or others. 

Currently StreamX workspace is forced to be stored on HDFS and should also support local FS or other FS, in the improved storage abstraction shot is to rely on workspace URI for automatic determination.

For example, **_"hdfs:///streamx/workspace"_** is automatically directed to HDFS, **_"/streamx/workspace/"_** or **_"local:///streamx/workspace"_** to local FS, **_"s3:///streamx/workspace/"**_ to S3.

**Comment by Al-assad** [comment]
StreamX users will have a requirement to use only Flink K8s schema features, such as issue #192. Their usage scenario for StreamX is to use it only as an ETL platform, with no Hadoop clusters included in the deployment scenario（with other OLAP downlinks such as ClickHouse, TiDB, etc.）

**Comment by wolfboys** [comment]
Yes, we need to remove the current mandatory dependence on Hadoop environment and make it optional. We can decide which environment dependence to use according to the different submission modes (horn application, k8s, local) selected by users. As far as Hadoop is concerned, there are HDFS based or S3 based ones. There may be some subtle differences and changes. As far as local mode is concerned, it is different from the implementation of Hadoop dependent environment, It's another kind of realization, and the details need to be discussed

**Comment by wolfboys** [comment]
In addition, Now cloud native is more and more popular, and there are more storage in this way, we need to discuss the storage of Hadoop on S3.

**Comment by Al-assad** [comment]
@wolfboys I think the focus is not on S3, but on allowing users to freely choose the resource storage of streamx or the freedom to decide on storage based on reality. 
as shown below:
![Untitled-2021-06-29-1245](https://user-images.githubusercontent.com/22493821/125230960-6e070a80-e30c-11eb-9768-e16b45d72d1d.png)

**Comment by datayangl** [comment]
Flink's design of  filesystem pretty match our goal, so I suggest  we can firstly go through the abstraction and implemention of flink, and make it configurable.
